src/setup/node.rs

```diff
@@ -260,8 +277,118 @@ impl Drop for Node {
+    async fn run_stateless_nodes_in_parallel() {
+        let mut builder = NodeBuilder::stateless().expect("Can't build a stateless node");
+        let mut nodes = Vec::<Node>::with_capacity(STATELESS_NODE_CNT);
+
+        for _ in 0..STATELESS_NODE_CNT {
+            let target = TempDir::new().expect("Can't build tmp dir");
+
+            let node = builder
+                .start(target.path(), NodeType::Stateless)
+                .await
+                .expect("Unable to start node");
+            nodes.push(node);
+        }
+
+        sleep(SLEEP).await;
+
+        for mut node in nodes {
+            node.stop().unwrap();
+        }
+    }
+
+    #[tokio::test]
+    async fn run_stateless_nodes_sequentially() {
+        let mut builder = NodeBuilder::stateless().expect("Can't build a stateless node");
+
+        for _ in 0..STATELESS_NODE_CNT {
+            let target = TempDir::new().expect("Can't build tmp dir");
+
+            let mut node = builder
+                .start(target.path(), NodeType::Stateless)
+                .await
+                .expect("Unable to start node");
+
+            sleep(SLEEP).await;
+            node.stop().unwrap();
+        }
+    }
+
+    #[tokio::test]
+    async fn run_stateful_nodes_sequentially() {
+        let mut builder = NodeBuilder::stateful().expect("Can't build a stateful node");
+
+        for _ in 0..STATEFUL_NODES_CNT {
+            let target = TempDir::new().expect("Can't build tmp dir");
+
+            let mut node = builder
+                .start(target.path(), NodeType::Stateful)
+                .await
+                .expect("Unable to start node");
+
+            sleep(SLEEP).await;
+            node.stop().unwrap();
+        }
+    }
+
+    #[tokio::test]
+    #[should_panic]
+    async fn run_too_many_stateful_nodes_sequentially() {
+        let mut builder = NodeBuilder::stateful().expect("Can't build a stateful node");
+
+        for _ in 0..TOO_MANY_STATEFUL_NODES_CNT {
+            let target = TempDir::new().expect("Can't build tmp dir");
+            let mut node = builder
+                .start(target.path(), NodeType::Stateful)
+                .await
+                .expect("Unable to start node");
+
+            sleep(SLEEP).await;
+            node.stop().unwrap();
+        }
+    }
+
+    #[tokio::test]
+    async fn run_stateful_nodes_in_parallel() {
+        let mut builder = NodeBuilder::stateful().expect("Can't build a stateful node");
+        let mut nodes = Vec::<Node>::with_capacity(STATEFUL_NODES_CNT);
+
+        for _ in 0..STATEFUL_NODES_CNT {
+            let target = TempDir::new().expect("Can't build tmp dir");
+
+            let node = builder
+                .start(target.path(), NodeType::Stateful)
+                .await
+                .expect("Unable to start node");
+            nodes.push(node);
+        }
+
+        sleep(SLEEP).await;
+
+        for mut node in nodes {
+            node.stop().unwrap();
+        }
+    }
```

Comment:
It's good to have these tests, but I would suggest marking them `ignored` as they don't really test for anything, just prove  that running nodes like this is possible.

Reply:
So is it okay to keep them alive and then tag them as `ignored` in the next commit?

Reply:
I don't mind, sure.